### PR TITLE
Add a progress stepper to onboarding flow

### PR DESCRIPTION
**Pull request by vaidehijoshi**
What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
✨ Adds the world's _cutest-ever_ progress stepper to the onboarding slides! ✨

The four main "interactive" steps of the onboarding flow now are represented by a little progress stepper that shows up in the navigation of the onboarding slides.

As a part of this PR, I have also cleaned up the onboarding JS tests as well. 💪 

## Related Tickets & Documents
Addresses the first half of https://github.com/thepracticaldev/dev.to/issues/6544. (I will address the skip/continue feature in a later PR! 😄)

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

To QA this, you'll want to step through the onboarding flow locally.

You'll want to check that the stepper does _not_ render for the intro slide:
Screen Shot 2020-04-22 at 12 11 16 PM

You'll also want to make sure that the stepper changes based on which slide you're looking at!

For example, the first slide looks like this:
Screen Shot 2020-04-22 at 12 11 26 PM

And the last slide looks like this:
Screen Shot 2020-04-22 at 12 11 38 PM

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] ~no documentation needed~ **I added inline docs!**

## What gif best describes this PR or how it makes you feel?

glittery computer face

### CODE REVIEW

**Comment by vaidehijoshi** [comment]
Hm, is it possible to turn off Code Climate for `propTypes = {` definitions? I don't think we can really "refactor" that.

I also could use help from someone in manually approving Code Climate because it's only other complaint is the size of a `render` function, which I feel like, personally, is good to know about...but probably shouldn't block merging?

File: app/javascript/onboarding/__tests__/Onboarding.test.jsx [hunk 1]
```diff
@@ -83,7 +102,7 @@ describe('<Onboarding />', () => {
     });
 
   describe('IntroSlide', () => {
-    const introSlideIndex = 0;
+    const slideIndex = slides.indexOf('IntroSlide');
```
Comment (Zhao-Andy) [inline app/javascript/onboarding/__tests__/Onboarding.test.jsx:105]: Love this refactor 👍

File: app/javascript/onboarding/Onboarding.jsx [hunk 2]
```diff
@@ -46,6 +49,7 @@ export default class Onboarding extends Component {
         currentSlide: nextSlide,
       });
     } else {
+      // Redirect to the main feed at the end of onboarding.
```
Comment (maestromac) [inline app/javascript/onboarding/Onboarding.jsx:52]: Thanks for adding this!

Comment (vaidehijoshi) [reply app/javascript/onboarding/Onboarding.jsx:52]: You got it! 😉

**Pull request merged**